### PR TITLE
Adds the "infamous triplet" and exports default and names exports in TypeScript types

### PR DESCRIPTION
**Pull request by fox1t**
This PR is the result of the ongoing talks about supporting all import contexts for the fastify ecosystem. We can use it as a case study for other core plugins.

What is important to note is that it is not enough to add this code only at JS, since the exported typings need to be updated too.

- Adds the "infamous triplet" in `module.exports`
- Replaces namespace export with "infamous triplet" in TypeScript typings
- Adds test for all TS imports
- Adds test for ESM context

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "prepare": "node prepare-swagger-ui",
-    "test": "standard && tap --100 test/*.js && npm run typescript",
+    "test": "standard && tap --no-esm --100 test/*.js test/esm/*.js && npm run typescript",
```
Comment (mcollina) [inline package.json:8]: https://github.com/fastify/fastify/blob/master/.taprc < use this instead.

Comment (fox1t) [reply package.json]: Done

File: index.js [hunk 2]
```diff
@@ -22,7 +22,11 @@ function fastifySwagger (fastify, opts, next) {
...
 
-module.exports = fp(fastifySwagger, {
+const fastifySwaggerPlugin = fp(fastifySwagger, {
   fastify: '>=3.x',
   name: 'fastify-swagger'
 })
+
+module.exports = fastifySwaggerPlugin
+module.exports.default = fastifySwaggerPlugin
+module.exports.fastifySwagger = fastifySwaggerPlugin
```
Comment (mcollina) [inline index.js:32]: I don't think these last two are needed with the new fastify-plugin release

Comment (fox1t) [reply index.js]: I confirm that we don't need it. Removed.

**Pull request merged**